### DISCUSSION TITLE
Quicker initial access to DLS data

### DISCUSSION THREAD

**Discussion by daviestar** [AlbrechtL/welle.io#830] [Q&A]
Hi there,

I've been experimenting with this library and I've got it working for my needs, thanks very much!

Using the command `welle-cli -c 12B -C 1 -w 7979` to start welle, then polling the URL `http://localhost:7979/mux.json`, I can retrieve the `Now playing: <track> by <artist>` DLS information, which I can then use to retrieve cover art for the currently playing track... great!

`12B` is the BBC National DAB ensemble in the UK:

![Screenshot 2024-12-14 at 16 23 16](https://github.com/user-attachments/assets/3ccea978-56b4-4aaf-abf5-bf551357e1a9)

If I leave the `welle-cli` process running 24/7, the currently playing track can be retrieved near-instantly when turning the radio on, however I'm wondering if I can optimise the system so to start the `welle-cli` process when turning on the radio and shut it down when the radio is off, but it can take a few minutes before DLS info seems to come through for my particular station from the ensemble.  I'm only interested in BBC Radio6Music which always seems to be one of the last to update with DLS data.

Is there a way I can speed up the initial DLS data coming through? I thought perhaps I could "tune in" to just the this station somehow instead of the whole ensemble?

Thanks, Dave

**Comment by AlbrechtL** [comment]
@mpbraendli Can you answer @daviestar, please?

**Comment by mpbraendli** [comment]
Hi Dave, if you're only interested in a single service, don't use `-C` nor `-D` but select your programme with the `-p` option. Then it'll only decode that one, and that should be faster.

**Comment by dmca-glasgow** [comment]
Hello @mpbraendli thanks for your help.  Sorry for the delay, 10 months later I'm back to this...

I'm not confident what value exactly to put for `-p`, but when I try any of these combinations:

```
welle-cli -w 7979 -c 12B -p "BBC Radio6Music"
welle-cli -w 7979 -c 12B -p "BBC R6M"
welle-cli -w 7979 -c 12B -p BBC Radio6Music
welle-cli -w 7979 -c 12B -p BBC R6M
welle-cli -w 7979 -c 12B -p "BBC Radio6Music (BBC R6M)"
```
The server starts, but no data seems to come through from the radio and I don't see `MP2Decoder: using decoder 'NEON64'.` in the console.  Instead I get `ofdm-processor: SyncOnPhase failed`.

Any ideas?  Thanks.

**Reply by dmca-glasgow** [reply]
Here is the output for the station from `mux.json`:

```js
{
  audiolevel: { left: 16743, right: 19479, time: 1760205753 },
  channels: 2,
  components: [
    {
      ascty: 'DAB',
      caflag: false,
      componentnr: 0,
      dscty: null,
      label: {
        fig2charset: 'Undefined',
        fig2label: '',
        fig2rfu: false,
        label: '',
        shortlabel: ''
      },
      primary: true,
      scid: null,
      subchannel: {
        bitrate: 112,
        cu: 84,
        language: 9,
        languagestring: 'English',
        protection: 'UEP 3',
        sad: 668,
        subchid: 11
      },
      transportmode: 'audio'
    }
  ],
  dls: {
    label: 'We welcome Lack of Afro as he talks about his latest and possibly final album.',
    lastchange: 0,
    time: 1760205747
  },
  errorcounters: { aacerrors: 0, frameerrors: 0, rserrors: 0, time: 1760205747 },
  label: {
    fig2charset: 'Undefined',
    fig2label: '',
    fig2rfu: false,
    label: 'BBC Radio6Music ',
    shortlabel: 'BBC R6M'
  },
  language: 9,
  languagestring: 'English',
  mode: 'MPEG 1.0 Layer II, 48 kHz Joint Stereo @ 112 kbit/s',
  mot: { lastchange: 0, time: 0 },
  programType: 11,
  ptystring: 'Rock Music',
  samplerate: 48000,
  sid: '0xc22b',
  url_mp3: '/mp3/0xc22b',
  xpaderror: { haserror: false }
}
```

**Reply by mpbraendli** [reply]
Hi, erm, I'm a bit confused. What doesn't work?
I see your `mux.json` with the DLS, and I also see valid audio levels, and so I suppose the audio decoding works too...

**Reply by dmca-glasgow** [reply]
Sorry for the confusion, the `mux.json` output is from when I use the working command:

```
welle-cli -c 12B -C 1 -w 7979
```

Wait a few minutes for everything to load in, and inspect the output for the station BBC Radio6Music.

My problem is when I try to follow your advice and use the `-p` command.  In that case, it's as I say above, the web server starts but there is no radio communication.  My guess is I'm not putting in the right value for `-p`, so I pasted all the details I have from `mux.json`, as I appreciate it's hard for you to debug if you're not in the UK.

Thanks.
